### ISSUE TITLE
No locale changes when switching language

### ISSUE THREAD

**Issue by shadowrider-pl** [jhipster/generator-jhipster#6976]
##### **Overview of the issue**

I can't get currency and date format for specific language when I change it in navbar menu.
Like: <td>{{product.sellPrice | currency}}</td>
- it's always dollar symbol.
In AngularJS I used before it was working out of the box as I remember.

##### **JHipster configuration**

{
  "generator-jhipster": {
    "promptValues": {
      "packageName": 
      "nativeLanguage": "en"
    },
    "jhipsterVersion": "4.13.1",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "clusteredHttpSession": false,
    "websocket": "spring-websocket",
    "databaseType": "sql",
    "devDatabaseType": "mariadb",
    "prodDatabaseType": "mariadb",
    "searchEngine": "elasticsearch",
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSocialSignIn": false,
    "enableSwaggerCodegen": false,
    "clientFramework": "angularX",
    "useSass": true,
    "clientPackageManager": "yarn",
    "applicationType": "monolith",
    "testFrameworks": [
      "gatling"
    ],
    "jhiPrefix": "jhi",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": [
      "en",
      "fr",
      "de",
      "pl",
      "pt-pt",
      "es",
      "ua"
    ]
  }
}

##### **Browsers and Operating System**

Ubuntu, all browsers.

**Comment by deepu105** [comment]
Can you please run `jhipster info` and copy paste the output here otherwise we would close the issue as this doesnt follow our issue guidelines

**Comment by shadowrider-pl** [comment]
$ jhipster info
Using JHipster version installed locally in current project's node_modules
Executing jhipster:info
Options: 
Welcome to the JHipster Information Sub-Generator

##### **JHipster Version(s)**

```
barfitter@0.0.0 /etc/SSD_documents/barfitter
├── generator-jhipster@4.13.1 
└── UNMET PEER DEPENDENCY popper.js@^1.12.3

```


##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**


<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "finbarre.barfitter",
      "nativeLanguage": "en"
    },
    "jhipsterVersion": "4.13.1",
    "baseName": "barfitter",
    "packageName": "finbarre.barfitter",
    "packageFolder": "finbarre/barfitter",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "clusteredHttpSession": false,
    "websocket": "spring-websocket",
    "databaseType": "sql",
    "devDatabaseType": "mariadb",
    "prodDatabaseType": "mariadb",
    "searchEngine": "elasticsearch",
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSocialSignIn": false,
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "replaced-by-jhipster-info",
    "clientFramework": "angularX",
    "useSass": true,
    "clientPackageManager": "yarn",
    "applicationType": "monolith",
    "testFrameworks": [
      "gatling"
    ],
    "jhiPrefix": "jhi",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": [
      "en",
      "fr",
      "de",
      "pl",
      "pt-pt",
      "es",
      "ua"
    ]
  }
}
</pre>
</details>


##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity Vat (vat) {
  description String required,
  rate BigDecimal required,
  active Boolean
}
entity Product (product) {
  name String required,
  purchPriceNet BigDecimal,
  sellPriceGross BigDecimal required,
  active Boolean,
  purchPriceGross BigDecimal,
  purchVatValue BigDecimal,
  sellPriceNet BigDecimal,
  sellVatValue BigDecimal
}
entity Category (category) {
  name String required,
  parentId Integer,
  active Boolean
}
entity ProductDelivered (product_delivered) {
  name String,
  deliveryDate LocalDate,
  quantity Integer,
  purchPriceGross BigDecimal,
  sellPriceGross BigDecimal required,
  purchPriceNet BigDecimal,
  purchVatValue BigDecimal,
  sellPriceNet BigDecimal,
  sellVatValue BigDecimal
}
entity ProductOnStock (product_on_stock) {
  deliveryDate LocalDate,
  quantity Integer,
  purchPriceNet BigDecimal,
  purchPriceGross BigDecimal,
  purchVatValue BigDecimal,
  sellPriceNet BigDecimal,
  sellPriceGross BigDecimal required,
  sellVatValue BigDecimal
}
entity OrderOpened (order_opened) {
  total BigDecimal,
  comment String,
  openingTime ZonedDateTime
}
entity Payment (payment) {
  description String required,
  active Boolean
}
entity ProductType (product_type) {
  description String required,
  active Boolean
}
entity OrderedProductStatus (ordered_product_status) {
  description String required,
  active Boolean
}
entity Desk (desk) {
  description String required,
  active Boolean,
  parentId Long
}
entity ProductOrdered (product_ordered) {
  orderedTime ZonedDateTime required,
  acceptedTime ZonedDateTime,
  finishedTime ZonedDateTime,
  takenTime ZonedDateTime,
  quantity Integer,
  comment String,
  purchPriceNet BigDecimal,
  purchPriceGross BigDecimal,
  purchVatValue BigDecimal,
  sellPriceNet BigDecimal,
  sellPriceGross BigDecimal,
  sellVatValue BigDecimal,
  difference BigDecimal,
  deliveryDate LocalDate,
  orderPosition Integer,
  sendTime ZonedDateTime
}
entity OrderClosed (order_closed) {
  openingTime ZonedDateTime required,
  closingTime ZonedDateTime required,
  total BigDecimal,
  comment String
}
entity ProductSold (product_sold) {
  orderedTime ZonedDateTime required,
  acceptedTime ZonedDateTime,
  finishedTime ZonedDateTime,
  takenTime ZonedDateTime,
  quantity Integer,
  comment String,
  purchPriceNet BigDecimal,
  purchPriceGross BigDecimal,
  purchVatValue BigDecimal,
  sellPriceNet BigDecimal,
  sellPriceGross BigDecimal,
  sellVatValue BigDecimal,
  difference BigDecimal,
  deliveryDate LocalDate,
  sendTime ZonedDateTime
}
entity Cashup (cashup) {
  barmanLoginTime ZonedDateTime required,
  cashupTime ZonedDateTime,
  startCash BigDecimal,
  endCash BigDecimal,
  totalSale BigDecimal,
  cashTakenByManager BigDecimal,
  cashTakenByBoss BigDecimal,
  comment String
}
entity PaymentToCashup (payment_to_cashup) {
  totalPayment BigDecimal
}
entity Xsell (xsell)
entity Bestseller (bestseller)
entity Favorite (favorite)
entity RestaurantConfiguration (configuration) {
  confText1 String,
  confText2 String,
  confText String,
  confDate1 ZonedDateTime,
  confDate2 ZonedDateTime,
  confInt1 Integer,
  confInt2 Integer,
  confBoolean1 Boolean,
  confBoolean2 Boolean,
  confBigDecimal1 BigDecimal,
  confBigDecimal2 BigDecimal,
  confDouble1 Double,
  confDouble2 Double
}
entity Restaurant (restaurant) {
  name String,
  country String,
  address String,
  zipCode String,
  city String,
  vatNumber String,
  licence LocalDate
}
entity UserToRestaurant (user_to_restaurant)

relationship ManyToOne {
  Vat{restaurant} to Restaurant,
  Product{productSellPriceRate} to Vat,
  Product{productPurchPriceRate} to Vat,
  Product{productType} to ProductType,
  Product{category} to Category,
  Product{restaurant} to Restaurant,
  Category{restaurant} to Restaurant,
  ProductDelivered{product} to Product,
  ProductDelivered{category} to Category,
  ProductDelivered{productDeliveredPurchPriceRate} to Vat,
  ProductDelivered{productDeliveredSellPriceRate} to Vat,
  ProductDelivered{productType} to ProductType,
  ProductDelivered{restaurant} to Restaurant,
  ProductOnStock{product} to Product,
  ProductOnStock{restaurant} to Restaurant,
  OrderOpened{payment} to Payment,
  OrderOpened{desk} to Desk,
  OrderOpened{barman(login)} to User,
  OrderOpened{restaurant} to Restaurant,
  Payment{restaurant} to Restaurant,
  Desk{restaurant} to Restaurant,
  ProductOrdered{order} to OrderOpened,
  ProductOrdered{orderedProductStatus} to OrderedProductStatus,
  ProductOrdered{productOrderedPurchPriceRate(description)} to Vat,
  ProductOrdered{productOrderedSellPriceRate(description)} to Vat,
  ProductOrdered{product(name)} to Product,
  ProductOrdered{chef(login)} to User,
  ProductOrdered{restaurant} to Restaurant,
  OrderClosed{cashupDay} to Cashup,
  OrderClosed{desk} to Desk,
  OrderClosed{payment} to Payment,
  OrderClosed{barman(login)} to User,
  OrderClosed{restaurant} to Restaurant,
  ProductSold{product} to Product,
  ProductSold{order} to OrderClosed,
  ProductSold{productSoldPurchPriceRate(description)} to Vat,
  ProductSold{productSoldSellPriceRate(description)} to Vat,
  ProductSold{chef(login)} to User,
  ProductSold{restaurant} to Restaurant,
  Cashup{cashupUser} to User,
  Cashup{openingUser} to User,
  Cashup{restaurant} to Restaurant,
  PaymentToCashup{cashup} to Cashup,
  PaymentToCashup{payment} to Payment,
  Xsell{fromCategory} to Category,
  Xsell{toCategory} to Category,
  Xsell{restaurant} to Restaurant,
  Bestseller{product(name)} to Product,
  Bestseller{restaurant} to Restaurant,
  Favorite{product(name)} to Product,
  Favorite{restaurant} to Restaurant,
  RestaurantConfiguration{restaurant} to Restaurant,
  UserToRestaurant{restaurant(name)} to Restaurant,
  UserToRestaurant{user(login)} to User
}

paginate Vat, Product, Category, ProductDelivered, ProductOnStock, OrderOpened, Payment, Desk, ProductOrdered, OrderClosed, ProductSold, Cashup, PaymentToCashup, Xsell, Bestseller, Favorite, RestaurantConfiguration, Restaurant, UserToRestaurant with pagination
service RestaurantConfiguration with serviceClass

</pre>
</details>


##### **Environment and Tools**

java version "1.8.0_151"
Java(TM) SE Runtime Environment (build 1.8.0_151-b12)
Java HotSpot(TM) 64-Bit Server VM (build 25.151-b12, mixed mode)

git version 1.9.1

node: v8.9.4

npm: 3.10.9

bower: 1.7.9

gulp:
[10:02:22] CLI version 1.2.2
[10:02:22] Local version 3.9.1

yeoman: 2.0.0

yarn: 1.3.2

Congratulations, JHipster execution is complete!

**Comment by gmarziou** [comment]
JHipster does not really manage locales, only languages without countries so what would you expect as currency for Spanish?

**Comment by shadowrider-pl** [comment]
Currency for Spanish should be Euro, not dollar.
I've just created 2 simple apps - one with AngularJS, second with Angular 5.
One with AngularJS looks good:
en:
![test es opera_070](https://user-images.githubusercontent.com/9105435/34890410-15bbf0ee-f7d2-11e7-99d5-f13651adf3a1.png)

pl:
![test es opera_071](https://user-images.githubusercontent.com/9105435/34890426-2d6e2518-f7d2-11e7-9593-1c3bbd390b4b.png)
Currency and date format is correct.

Here is Angular 5 version:
en:
![test es opera_068](https://user-images.githubusercontent.com/9105435/34890464-5368ca34-f7d2-11e7-8244-d8b35fd16f70.png)

pl:
![test es opera_067](https://user-images.githubusercontent.com/9105435/34890472-5c5f72b4-f7d2-11e7-8e0a-95fbb7b6a3aa.png)
Currency and date format is the same like in en.

**Comment by gmarziou** [comment]
Maybe my point was not clear enough, let me try again.

Currency for Spain should be euro but there's no currency for Spanish language: not the same currency for Argentina, Mexico, Columbia or any other Spanish speaking countries.'es' for JHipster means Spanish not Spain, the only way to to manage this would be to use Locales on client side like 'es_es'.

**Comment by deepu105** [comment]
We do not handle locale, AngularJS has built in pipes which were handling those automatically, we didn't do anything there. Apparently Angular doesn't do that anymore. We only handle i18n language.

**Comment by shadowrider-pl** [comment]
That's sad. Here is how I solve it:
https://codefitter2.blogspot.com/2018/01/how-to-change-date-format-and-currency.html
You can consider to include it in JHipster.

**Comment by deepu105** [comment]
It's not easy as this would mean we also would have to maintain nad handle locale. SO unless someone wants to PR this, we dont have the bandwidth. @shadowrider-pl you are welcome to contribute a PR if you like

**Comment by darksidersD** [comment]
@shadowrider-pl  
thanks a lot, it work well for me, but every time in need change the current page to see the change, 
is there any solutions? i mean to work like translate pipe
